### ISSUE TITLE
test_metadata_fixup_warning is failing due to metadata overwrites

### ISSUE THREAD

**Issue by QuLogic** [jbarlow83/OCRmyPDF#705]
**Describe the bug**
I'm updating the Fedora package to 11.4.2 and the following test fails:

```
_________________________ test_metadata_fixup_warning __________________________
[gw0] linux -- Python 3.9.1 /usr/bin/python3
resources = PosixPath('/builddir/build/BUILD/ocrmypdf-11.4.2/tests/resources')
outdir = PosixPath('/tmp/pytest-of-mockbuild/pytest-0/popen-gw0/test_metadata_fixup_warning0')
caplog = <_pytest.logging.LogCaptureFixture object at 0x7f828285d2b0>
    def test_metadata_fixup_warning(resources, outdir, caplog):
        options = get_parser().parse_args(
            args=['--output-type', 'pdfa-2', 'graph.pdf', 'out.pdf']
        )
    
        copyfile(resources / 'graph.pdf', outdir / 'graph.pdf')
    
        context = PdfContext(
            options, outdir, outdir / 'graph.pdf', None, get_plugin_manager([])
        )
        metadata_fixup(working_file=outdir / 'graph.pdf', context=context)
        for record in caplog.records:
>           assert record.levelname != 'WARNING'
E           assert 'WARNING' != 'WARNING'
E            +  where 'WARNING' = <LogRecord: pikepdf.models.metadata, 30, /usr/lib64/python3.9/site-packages/pikepdf/models/metadata.py, 680, "Update to {http://ns.adobe.com/pdf/1.3/}Producer will be overwritten because metadata was opened with set_pikepdf_as_editor=True">.levelname
tests/test_metadata.py:321: AssertionError
------------------------------ Captured log call -------------------------------
WARNING  pikepdf.models.metadata:metadata.py:680 Update to {http://ns.adobe.com/pdf/1.3/}Producer will be overwritten because metadata was opened with set_pikepdf_as_editor=True
WARNING  pikepdf.models.metadata:metadata.py:680 Update to {http://ns.adobe.com/xap/1.0/}MetadataDate will be overwritten because metadata was opened with set_pikepdf_as_editor=True
WARNING  pikepdf.models.metadata:metadata.py:680 Update to {http://ns.adobe.com/pdf/1.3/}Producer will be overwritten because metadata was opened with set_pikepdf_as_editor=True
```

**To Reproduce**
Run test suite.

**Expected behavior**
Tests pass.

**System (please complete the following information):**
 - OS: Fedora Rawhide
 - Python version: 3.9
 - OCRmyPDF version: 11.4.2

**Additional context**
Pikepdf 2.2.2 if that's relevant.

**Comment by QuLogic** [comment]
Updated to pikepdf 2.2.3 and now it warns only once about `Producer`.

**Comment by jbarlow83** [comment]
Thanks, should be fully fixed in 2.2.4.

**Comment by jbarlow83** [comment]
Fixed in ocrmypdf 11.4.3 and pikepdf 2.2.4
